aws/container-linux/kubernetes/security.tf

```diff
@@ -14,11 +14,21 @@ resource "aws_security_group" "controller" {
 resource "aws_security_group_rule" "controller-ssh" {
   security_group_id = "${aws_security_group.controller.id}"
 
-  type        = "ingress"
-  protocol    = "tcp"
-  from_port   = 22
-  to_port     = 22
-  cidr_blocks = ["0.0.0.0/0"]
+  type                     = "ingress"
+  protocol                 = "tcp"
+  from_port                = 22
+  to_port                  = 22
+  source_security_group_id = "${aws_security_group.worker.id}"
+}
+
+resource "aws_security_group_rule" "controller-ssh-self" {
```

Comment:
I don't think this is necessary, controller instances should not be initiating SSH connections to other controller instances

Reply:
It captures what we have right now. Looking at the AWS console I see a controller<->controller rule. That said, I am totally happy to delete things we do not need. I'll remove this.

Reply:
For which group? Looking @ spyro-controller I see only one rule for port 22, using an 0.0.0.0/0 CIDR. For lots of other things we do need controller <-> controller communication, e.g. etcd peers. Just not SSH.

Reply:
Sorry... I reworded the comment.

Reply:
Gotcha. That's why I left comments even for things that are needed. Wanted to highlight that the "allow SSH from worker instances" rule targets tunnelbox. If we were just using a bastion, we'd just have to allow 22 where the source group is bastion-internal.

Reply:
Removed.